cache/cache.go

```diff
@@ -0,0 +1,87 @@
+package cache
+
+import (
+	"github.com/mediocregopher/radix/v3"
+	"log"
+	"strconv"
+	"time"
+)
+
+const (
+	ZADD = "ZADD"
+	ZRANGEBYSCORE = "ZRANGEBYSCORE"
+	ZREMRANGEBYSCORE = "ZREMRANGEBYSCORE"
+)
+
+type CacheService interface {
+	// Insert new entities into sorted sets with the offset number as the score
+	Create(key string, delta string, score int64) error
+	//Fetch a range of offsets using a specified offset number as the starting offset
+	Read(key string, offset int64) ([]string, error)
+	//Delete a range of offsets from the sorted sets using matching timestamps.
+	Delete(key string) error
+}
+
+type RedisCacheService struct {
+	pool *radix.Pool
+	expiryInSeconds time.Duration
+}
+
+func NewRedisCacheService(network string, url string, size int, expiryInSeconds int64) CacheService {
+
+	pool, err := radix.NewPool(network, url, size)
+	if err != nil {
+		panic(err)
+	}
+	return &RedisCacheService{
+		pool: pool,
+		expiryInSeconds: time.Duration(expiryInSeconds),
+	}
+}
+
+func (r RedisCacheService) Create(key string, delta string, score int64) error {
```

Comment:
Am a little concerned this approach may cause duplicate offsets to be added to the second sorted set - is it feasible to replace this with an ADD + EXPIRE?

Reply:
EXPIRE works at key level. Currently all data is stored in the same key. We could create a new key for each delta using the offset as part of the key. Not sure whether this would complicate reading the data. Looking into it.

Reply:
Refactored so that each delta is stored in as a Redis string value with it's own key.
An expiry time is set on the key which Redis automatically handles so no need for manual delete.
The offsets are stored in a sorted set with the offset as key and the corresponding delta key at its value.
The Read() method:
1) Retrieves a list of keys for the deltas from the sorted set based on the given offset
2) Reads and returns the deltas for each delta key